### PR TITLE
Bumped actions versions

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Similar to changes already made on 1.x (other than one change, pending at #1669).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     php-tests:
         name: Tests - PHP ${{ matrix.php }} ${{ matrix.dependency-version }}
-        runs-on: ubuntu-latest
+        runs-on: ubuntu-22.04
```
Comment (mnapoli) [inline .github/workflows/ci.yml:15]: What do we gain again by fixing the version? We'll have to update this eventually 😬

Comment (GrahamCampbell) [reply .github/workflows/ci.yml:15]: We gain control over when this happens, just like we control the major versions on the other action things.

Comment (mnapoli) [reply .github/workflows/ci.yml:15]: Right but actions can have BC breaks on major versions.

I don't see any way the Ubuntu version would break the CI?

Comment (GrahamCampbell) [reply .github/workflows/ci.yml:15]: Ubuntu version typically does have breaking changes, and different actions may not be compatible yet.

Comment (GrahamCampbell) [reply .github/workflows/ci.yml:15]: GitHub also doesn't upgrade everything at once, doing a rolling move per repo, so it's kinda janky to sometimes have one version, and sometimes another.

**Pull request merged**